### PR TITLE
CLD-4642 Add support for SLO metric checks while in soaking state

### PR DESCRIPTION
**Pull request by stylianosrigas**
#### Summary
-  Add support for SLO metric checks while in soaking state

### CODE REVIEW

File: internal/elrond/elrond_soak.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+				return errors.Errorf("SLO service %s is showing a high API error rate. Soaking failed, stopping release", sloService)
+			}
+		} else {
+			logger.Info("SLO metric checks looking good...")
+		}
+	}
+
+	if err != nil {
+		return errors.Wrap(err, "failed to query thanos")
+	}
```
Comment (angeloskyratzakos) [inline internal/elrond/elrond_soak.go:35]: this should be higher otherwise the for loop will break, right?

File: internal/elrond/elrond_soak.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+	"github.com/sirupsen/logrus"
+)
+
+func checkSLOs(ring *model.Ring, thanosURL string, logger *logrus.Entry) error {
+	results, err := querySLOMetrics(ring, thanosURL, time.Now(), logger)
+	for _, result := range results {
+		if len(result) > 0 {
+			for _, rawMetric := range result {
+				sloService, ok := rawMetric.Metric["slo_service"]
+				if !ok {
```
Comment (angeloskyratzakos) [inline internal/elrond/elrond_soak.go:23]: renaming the variable `ok` to make more sense, like `issue`, `highSLO`, `errorsSLO` so when checking `if !issue ... continue`

File: internal/elrond/elrond_soak.go (hunk 1 above)
Comment (stylianosrigas) [reply internal/elrond/elrond_soak.go]: > this should be higher otherwise the for loop will break, right?

Nice catch thanks dude

**Review by angeloskyratzakos — APPROVED** [review]
nice addition @stylianosrigas LGTM! 🎉

File: internal/elrond/elrond_soak.go (hunk 2 above)
Comment (angeloskyratzakos) [reply internal/elrond/elrond_soak.go:27]: So you are checking if it is invalid? so if !`invalid` continue?

Comment (stylianosrigas) [reply internal/elrond/elrond_soak.go]: > So you are checking if it is invalid? so if !`invalid` continue?

Actually if the entry exists. Changed it to !exists ;)

**Pull request merged**